### PR TITLE
Overload '<' and '>' operators in `LazyUUIDTaskSet`

### PR DESCRIPTION
**Pull request by RonMcKay**
In #96 the request was to overload the '<' and '>' operators in `LazyUUIDTaskSet`. I did that in this PR by adding the methods `isstrictsubset` and `isstrictsuperset` to the class.
If I understood the issue correct this was everything that needs to be implemented for that. If I missed something I am happy for a remark.

### CODE REVIEW

File: tasklib/lazy.py [hunk 1]
```diff
@@ -165,12 +165,24 @@ def __le__(self, other):
...
+        return self.isstrictsuperset(other)
+
     def issubset(self, other):
         return all([task in other for task in self])
 
     def issuperset(self, other):
         return all([task in self for task in other])
 
+    def isstrictsubset(self, other):
+        return len(set(t['uuid'] for t in other) - self._uuids) > 0
```
Comment (tbabej) [inline tasklib/lazy.py:181]: I think this is actually not a correct definition of a strict subset criterium. Consider the following two sets:
```python
>>> A = {1,2,3}
>>> B = {1,2,4}
>>> B-A
{4}
>>> len(B-A) > 0
True
```

however, the sets are actually incomparable. Same problem holds for the implementation of `isstrictsuperset` below.

For a fix, I'd suggest using `<` and `>` operators for the underlying sets which we're already constructing, like this:
```python
return self._uuids < set(t['uuid'] for t in other)
```

Alternatively, the already existing implementations of `<=` and `>=` could be reused, i.e. the `<` operator could be defined as:
```python
return not self >= other
```

**Review by tbabej — CHANGES_REQUESTED** [review]
Otherwise looks good! I also think we should add a tests covering the behaviour of these new operators, as we can see getting all the cornercases working right can be tricky :slightly_smiling_face:

File: tasklib/lazy.py (hunk 1 above)
Comment (RonMcKay) [reply tasklib/lazy.py:181]: Yes, you are absolutely right. I was thinking to fast there 😄
But your second proposed variant is also not working:

```python
>>> A = {1,2,3}
>>> B = {1,2,4}
>>> all([i in A for i in B])  # current implementation of issuperset(self, other)
False
>>> not all([i in A for i in B])
True
```

So I would go with your first suggestion and just use the comparison operator of the `set` class.
In terms of readability it would be nice to also use the `set` operators for `__le__` and `__ge__`. What do you think?

Comment (tbabej) [reply tasklib/lazy.py:181]: >  But your second proposed variant is also not working

Ah yes, that was a brain fart on my end. These UUID sets are not well-ordered, in other words, it does not hold that `A<=B` implies `B<A`.

> So I would go with your first suggestion and just use the comparison operator of the set class.

Sounds good.

>  In terms of readability it would be nice to also use the set operators for __le__ and __ge__. What do you think?

I think this is good refactoring. Existing tests should also help us ensure we don't introduce bugs doing this.

> Edit: Should we also remove the is(strict){subset,superset} methods and implement the functionality directly inside __le__, __ge__, __lt__, __gt__? They do not bring any functionality in that case.

We can't remove existing API, meaning `issubset` and `issuperset` need to stay around, but we don't need to introduce `isstrictsubset` and `isstrictsuperset` (these two methods don't exist in the native `set` object's API either).

**Comment by RonMcKay** [comment]
I added some commits to address the discussion above. In summary the following changed:

- Implemented the `__lt__` and `__gt__` methods by using the comparison operators of `set`
- Refactored the `issubset` and `issuperset` methods by also using the comparison operators of `set`
- Added unittests for `__lt__`, `__gt__`, `__le__` and `__ge__`. Please let me know if I have forgotten a corner case.
- Fixed a wrong indentation in `tests.py`

All tests are passing for me.

**Review by tbabej — APPROVED** [review]
This all looks great, thanks for adjusting the PR and implementing exhaustive tests.
